lib/PuppeteerSharp/Request.cs

```diff
@@ -313,5 +320,11 @@ public async Task AbortAsync(RequestAbortErrorCode errorCode = RequestAbortError
 
         private Header[] HeadersArray(Dictionary<string, string> headers)
             => headers?.Select(pair => new Header { Name = pair.Key, Value = pair.Value }).ToArray();
+
+        private Header[] HeadersArray(List<KeyValuePair<string, string>> headers)
+            => headers?.Select(pair => new Header { Name = pair.Key, Value = pair.Value }).ToArray();
+
+        private bool ContainsKey(List<KeyValuePair<string, string>> headers, string key)
+           => headers?.Where(item => item.Key == key).ToList().Count > 0 ? true : false;
```

Comment:
If I may add an opinion - you could rewrite this `headers?.Where(item => item.Key == key).ToList().Count` as follows `headers?.Count(item => item.Key == key)`. There is no need to create new list object while we can use LINQ 😄